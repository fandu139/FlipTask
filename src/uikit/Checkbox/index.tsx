import { ICON_CHECK } from '../../assets/icon';
import Colors from '../../theme/colors';
import Icon from '../../uikit/Icon';
import React from 'react';
import { TouchableOpacity, StyleSheet, View, Text, StyleProp, TextStyle } from 'react-native';
import AppStyles from '../../theme/appStyles';

interface Props {
  disabled?: boolean;
  checked: boolean;
  onChangeValue: () => void;
  testID: string;
  accessibilityLabel?: string;
  label?: string;
  labelStyle?: StyleProp<TextStyle>;
}

const Checkbox: React.FC<Props> = ({
  disabled,
  checked,
  onChangeValue,
  testID,
  accessibilityLabel,
  label,
  labelStyle,
}: Props) => (
  <TouchableOpacity
    testID={testID}
    accessibilityLabel={accessibilityLabel || testID}
    disabled={disabled}
    onPress={onChangeValue}
    style={styles.container}
    hitSlop={AppStyles.HitSlopValue}
  >
    <View style={styles.checkbox}>
      {checked && <Icon name={ICON_CHECK} size={11} color={Colors.RED} />}
    </View>
    {label && <Text style={[styles.labelCheckbox, labelStyle]}>{label}</Text>}
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flexDirection: 'row',
  },
  checkbox: {
    width: 20,
    height: 20,
    borderColor: Colors.GRAYE0,
    borderWidth: 1,
    borderRadius: 3,
    justifyContent: 'center',
    alignItems: 'center',
  },
  labelCheckbox: {
    marginLeft: 10,
    width: '100%',
  },
});

Checkbox.defaultProps = {
  disabled: false,
  accessibilityLabel: undefined,
  label: undefined,
  labelStyle: undefined,
};

export default Checkbox;
