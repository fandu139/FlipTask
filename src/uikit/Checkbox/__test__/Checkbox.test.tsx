import Checkbox from '../../Checkbox';
import React from 'react';
import { render } from '@testing-library/react-native';

describe('Checkbox component test', () => {
  describe('Checkbox snapshot test', () => {
    it('should render correctly', () => {
      const tree = render(
        <Checkbox testID="check-box-default" checked onChangeValue={() => undefined} />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
