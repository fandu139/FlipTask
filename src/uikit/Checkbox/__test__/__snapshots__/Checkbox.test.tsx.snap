// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component test Checkbox snapshot test should render correctly 1`] = `
<View
  accessibilityLabel="check-box-default"
  accessible={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 15,
      "left": 15,
      "right": 15,
      "top": 15,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
  testID="check-box-default"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#E0E0E0",
        "borderRadius": 3,
        "borderWidth": 1,
        "height": 20,
        "justifyContent": "center",
        "width": 20,
      }
    }
  >
    <View
      accessibilityLabel="button-icon"
      accessible={true}
      focusable={false}
      hitSlop={
        Object {
          "bottom": 15,
          "left": 15,
          "right": 15,
          "top": 15,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="button-icon"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 11,
            },
            Object {
              "color": "#EF4941",
            },
            Object {
              "fontFamily": "ColectionIcon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        î¤ž
      </Text>
    </View>
  </View>
</View>
`;
